java/fury-core/src/main/java/org/apache/fury/serializer/AbstractObjectSerializer.java

```diff
@@ -984,14 +993,22 @@ public static class InternalFieldInfo {
     protected final String qualifiedFieldName;
     protected final FieldAccessor fieldAccessor;
     protected boolean nullable;
+    protected boolean trackingRef;
 
-    private InternalFieldInfo(Descriptor d, short classId) {
+    private InternalFieldInfo(Fury fury, Descriptor d, short classId) {
       this.typeRef = d.getTypeRef();
       this.classId = classId;
       this.qualifiedFieldName = d.getDeclaringClass() + "." + d.getName();
       this.fieldAccessor = d.getField() != null ? FieldAccessor.createAccessor(d.getField()) : null;
+      FuryField furyField = d.getFuryField();
       if (!typeRef.isPrimitive()) {
-        nullable = d.getFuryField() == null || d.getFuryField().nullable();
+        nullable = furyField == null || furyField.nullable();
+      }
+      if (fury.trackingRef()) {
+        trackingRef =
+            furyField != null
+                ? furyField.trackingRef()
```

Comment:
If `furyField.trackingRef()` but `fury.trackingRef` is false, then it should still be false